refactored/src/edx/entry_table.rs

```diff
@@ -0,0 +1,178 @@
+//! The Entry Table is a Encrypted Dictionary scheme (EDX). It is used to
+//! securely store chain metadata.
+//!
+//! It uses the AES256-GCM algorithm in order to encrypt its values and the
+//! KMAC256 algorithm in order to derive secure tokens from tags.
+
+use std::{
+    collections::{HashMap, HashSet},
+    hash::Hash,
+    marker::PhantomData,
+};
+
+use cosmian_crypto_core::{
+    kdf,
+    reexport::rand_core::CryptoRngCore,
+    symmetric_crypto::{aes_256_gcm_pure::KEY_LENGTH, key::Key, SymKey},
+};
+
+use super::structs::{EdxKey, EncryptedValue, Seed};
+use crate::{
+    edx::{DxEnc, EdxStore},
+    error::Error,
+    parameters::{SEED_LENGTH, TOKEN_LENGTH},
+};
+
+/// Implementation of the Entry Table EDX.
+pub struct EntryTable<
+    const VALUE_LENGTH: usize,
+    EdxError: std::error::Error,
+    Edx: EdxStore<[u8; TOKEN_LENGTH], EncryptedValue<VALUE_LENGTH>, EdxError>,
+> {
+    pub edx: Edx,
+    err: PhantomData<EdxError>,
+}
+
+impl<
+    const VALUE_LENGTH: usize,
+    EdxError: std::error::Error,
+    Edx: EdxStore<[u8; TOKEN_LENGTH], EncryptedValue<VALUE_LENGTH>, EdxError>,
+> EntryTable<VALUE_LENGTH, EdxError, Edx>
+{
+    pub fn new(edx: Edx) -> Self {
+        Self {
+            edx,
+            err: PhantomData,
+        }
+    }
+}
+
+impl<
+    const VALUE_LENGTH: usize,
+    EdxError: std::error::Error,
+    Edx: EdxStore<[u8; TOKEN_LENGTH], EncryptedValue<VALUE_LENGTH>, EdxError>,
+> DxEnc<SEED_LENGTH, KEY_LENGTH, TOKEN_LENGTH, VALUE_LENGTH, Error<EdxError>>
+    for EntryTable<VALUE_LENGTH, EdxError, Edx>
+{
+    type EncryptedValue = EncryptedValue<VALUE_LENGTH>;
+    type Item = u8;
+    type Key = EdxKey<KEY_LENGTH>;
+    type Seed = Seed<SEED_LENGTH>;
+    type Token = [u8; TOKEN_LENGTH];
+
+    fn setup() -> Self {
+        Self {
+            edx: Edx::setup(),
+            err: PhantomData,
+        }
+    }
+
+    fn gen_seed(&self, rng: &mut impl CryptoRngCore) -> Self::Seed {
```

Comment:
It is currently not possible as default type are not allowed in traits. I would need a trait defining a method `new()` for  `Self::Seed` and `from_seed()` for `Self::Key`.

Since the amount of duplicated code is not that big, I prefer to let it as is.